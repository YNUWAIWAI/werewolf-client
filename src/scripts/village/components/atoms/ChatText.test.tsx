import * as React from 'react'
import {
  mount,
  shallow
} from 'enzyme'
import ChatText from './ChatText'

describe('<ChatText />', () => {
  describe('text', () => {
    test('text', () => {
      const wrapper = shallow(
        <ChatText
          navigatable
          text="text"
        />
      )

      expect(wrapper.text()).toBe('text')
      expect(wrapper.html()).toMatchSnapshot()
    })
    test('>>1 text', () => {
      const wrapper = shallow(
        <ChatText
          navigatable
          text=">>1 text"
        />
      )

      expect(wrapper.text()).toBe('>>1 text')
      expect(wrapper.find('a[href="#message1"]').exists()).toBe(true)
      expect(wrapper.html()).toMatchSnapshot()
    })
    test('>>1text', () => {
      const wrapper = shallow(
        <ChatText
          navigatable
          text=">>1text"
        />
      )

      expect(wrapper.text()).toBe('>>1text')
      expect(wrapper.find('a[href="#message1"]').exists()).toBe(true)
      expect(wrapper.html()).toMatchSnapshot()
    })
    test('>>n text', () => {
      const wrapper = shallow(
        <ChatText
          navigatable
          text=">>n text"
        />
      )

      expect(wrapper.text()).toBe('>>n text')
      expect(wrapper.find('a[href="#messagen"]').exists()).toBe(false)
      expect(wrapper.html()).toMatchSnapshot()
    })
    test('>>1 >>2 >>3 text', () => {
      const wrapper = shallow(
        <ChatText
          navigatable
          text=">>1 >>2 >>3 text"
        />
      )

      expect(wrapper.text()).toBe('>>1 >>2 >>3 text')
      expect(wrapper.find('a[href="#message1"]').exists()).toBe(true)
      expect(wrapper.find('a[href="#message2"]').exists()).toBe(true)
      expect(wrapper.find('a[href="#message3"]').exists()).toBe(true)
      expect(wrapper.html()).toMatchSnapshot()
    })
    test('text\\n\\r\\ntext', () => {
      const wrapper = shallow(
        <ChatText
          navigatable
          text={'text\n\r\ntext'}
        />
      )

      expect(wrapper.text()).toBe('texttext')
      expect(wrapper.html()).toMatchSnapshot()
    })
    test('text>>>1\\ntext', () => {
      const wrapper = shallow(
        <ChatText
          navigatable
          text={'text>>1\ntext'}
        />
      )

      expect(wrapper.text()).toBe('text>>1text')
      expect(wrapper.find('a[href="#message1"]').exists()).toBe(true)
      expect(wrapper.html()).toMatchSnapshot()
    })
  })
  describe('navigatable', () => {
    test('true', () => {
      const wrapper = mount(
        <ChatText
          navigatable
          text=">>1 text"
        />
      )

      expect(wrapper.find('a').props().tabIndex).toBe(0)
      expect(wrapper.html()).toMatchSnapshot()
    })
    test('false', () => {
      const wrapper = mount(
        <ChatText
          navigatable={false}
          text=">>1 text"
        />
      )

      expect(wrapper.find('a').props().tabIndex).toBe(-1)
      expect(wrapper.html()).toMatchSnapshot()
    })
  })
})
