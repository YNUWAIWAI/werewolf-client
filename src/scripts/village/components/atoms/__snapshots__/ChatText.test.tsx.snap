// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render navigatable false 1`] = `"<div class=\\"vi--chat--text\\"><p class=\\"vi--chat--text--chunk\\"><a href=\\"#message1\\" tabindex=\\"-1\\">&gt;&gt;1</a> text</p></div>"`;

exports[`render navigatable true 1`] = `"<div class=\\"vi--chat--text\\"><p class=\\"vi--chat--text--chunk\\"><a href=\\"#message1\\" tabindex=\\"0\\">&gt;&gt;1</a> text</p></div>"`;

exports[`render text >>1 >>2 >>3 text 1`] = `"<div class=\\"vi--chat--text\\"><p class=\\"vi--chat--text--chunk\\"><a href=\\"#message1\\" tabindex=\\"0\\">&gt;&gt;1</a> <a href=\\"#message2\\" tabindex=\\"0\\">&gt;&gt;2</a> <a href=\\"#message3\\" tabindex=\\"0\\">&gt;&gt;3</a> text</p></div>"`;

exports[`render text >>1 text 1`] = `"<div class=\\"vi--chat--text\\"><p class=\\"vi--chat--text--chunk\\"><a href=\\"#message1\\" tabindex=\\"0\\">&gt;&gt;1</a> text</p></div>"`;

exports[`render text >>1text 1`] = `"<div class=\\"vi--chat--text\\"><p class=\\"vi--chat--text--chunk\\"><a href=\\"#message1\\" tabindex=\\"0\\">&gt;&gt;1</a>text</p></div>"`;

exports[`render text >>n text 1`] = `"<div class=\\"vi--chat--text\\"><p class=\\"vi--chat--text--chunk\\">&gt;&gt;n text</p></div>"`;

exports[`render text text 1`] = `"<div class=\\"vi--chat--text\\"><p class=\\"vi--chat--text--chunk\\">text</p></div>"`;

exports[`render text text>>>1\\ntext 1`] = `"<div class=\\"vi--chat--text\\"><p class=\\"vi--chat--text--chunk\\">text<a href=\\"#message1\\" tabindex=\\"0\\">&gt;&gt;1</a></p><p class=\\"vi--chat--text--chunk\\">text</p></div>"`;

exports[`render text text\\n\\r\\ntext 1`] = `"<div class=\\"vi--chat--text\\"><p class=\\"vi--chat--text--chunk\\">text</p><p class=\\"vi--chat--text--chunk\\">text</p></div>"`;
